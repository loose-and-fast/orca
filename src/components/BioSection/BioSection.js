import React from 'react';
import { Image } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import './BioSection.css';

const BioSection = ({ name, text, url }) => (
  <div className="BioSection">
    <h2>{name}</h2>
    <Image src={url} size="medium" rounded floated="left" />
    <p>{text}</p>
  </div>
);

BioSection.propTypes = {
  name: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default BioSection;
