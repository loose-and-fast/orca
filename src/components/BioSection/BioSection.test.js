import React from 'react'
import { shallow } from 'enzyme';

import BioSection from './BioSection'

describe('BioSection', () => {
  let component, props

  beforeEach(() => {
    props = { url: 'testUrl', name: 'testName', text: 'testText' }
    component = shallow(<BioSection {...props} />)
  })

  it('should', () => {
    expect(component).toMatchSnapshot()
  })
})