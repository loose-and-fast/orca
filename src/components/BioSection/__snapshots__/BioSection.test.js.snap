// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BioSection should 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BioSection
    name="testName"
    text="testText"
    url="testUrl"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          testName
</h2>,
        <Image
          as="img"
          floated="left"
          rounded={true}
          size="medium"
          src="testUrl"
          ui={true}
/>,
        <p>
          testText
</p>,
      ],
      "className": "BioSection",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "testName",
        },
        "ref": null,
        "rendered": "testName",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "img",
          "floated": "left",
          "rounded": true,
          "size": "medium",
          "src": "testUrl",
          "ui": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "testText",
        },
        "ref": null,
        "rendered": "testText",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            testName
</h2>,
          <Image
            as="img"
            floated="left"
            rounded={true}
            size="medium"
            src="testUrl"
            ui={true}
/>,
          <p>
            testText
</p>,
        ],
        "className": "BioSection",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "testName",
          },
          "ref": null,
          "rendered": "testName",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "img",
            "floated": "left",
            "rounded": true,
            "size": "medium",
            "src": "testUrl",
            "ui": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "testText",
          },
          "ref": null,
          "rendered": "testText",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
